### PR TITLE
feat: implement number field editing

### PR DESCRIPTION
**Pull request by mantariksh**
_Depends on #3695_

Implements field builder for number fields.

### CODE REVIEW

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  )
+
+  const watchedSelectedValidation = watch(
+    'ValidationOptions.selectedValidation',
+  )
+
+  const customValValidationOptions: RegisterOptions = useMemo(
+    () => ({
+      required: {
+        value: watchedSelectedValidation !== '',
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx:107]: nit+perf: it's recommended to use `getValues` inside register options: https://react-hook-form.com/api/useform/getvalues

this means this memoized object will not recreate on selected validation change

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+          selectedValidation: '' as const,
+          customVal: '' as const,
+        }
+      : {
+          selectedValidation: field.ValidationOptions.selectedValidation,
+          customVal:
+            field.ValidationOptions.customVal === null
+              ? ('' as const)
+              : field.ValidationOptions.customVal,
+        }
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx:51]: nit: simplify?

```ts
const nextValidationOptions = {
  selectedValidation: field.validationOptions.selectedValidation || '',
  customVal: !!field.validationOptions.selectedValidation && field.ValidationOptions.customVal || ''
}
```

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            control={control}
+            rules={customValValidationOptions}
+            render={({ field }) => (
+              <NumberInput
+                flex={1}
+                showSteppers={false}
+                {...field}
+                placeholder="Number of characters"
+                isDisabled={watchedSelectedValidation === ''}
+              />
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx:176]: will `NumberInput` also allow other numbers without like `e`? But I guess validation will filter it out due to min and max

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            name="ValidationOptions.customVal"
+            control={control}
+            rules={customValValidationOptions}
+            render={({ field }) => (
+              <NumberInput
+                flex={1}
+                showSteppers={false}
+                {...field}
+                placeholder="Number of characters"
+                isDisabled={watchedSelectedValidation === ''}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx:175]: nit: simplify to `!watchedSelectedValidation`?

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        value: 10000,
+        message: 'Cannot be more than 10000',
+      },
+    }),
+    [watchedSelectedValidation],
+  )
+
+  // Effect to clear validation option errors when selection limit is toggled off.
+  useEffect(() => {
+    if (watchedSelectedValidation === '') {
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx:124]: nit: simplify to `!watchedSelectedValidation`?

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber/EditNumber.tsx [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+  originalField: NumberFieldBase,
+): NumberFieldBase => {
+  const nextValidationOptions =
+    form.ValidationOptions.selectedValidation === ''
+      ? {
+          selectedValidation: null,
+          customVal: null,
+        }
+      : form.ValidationOptions
+  return extend({}, originalField, form, {
```
Comment (timotheeg) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber/EditNumber.tsx:64]: is `extend()` overreaching here? What inherited properties do we need from `originalField` and `form`?

**Review by timotheeg — APPROVED** [review]
lgtm 👍

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber/EditNumber.tsx (hunk 6 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber/EditNumber.tsx:64]: it's fine I think, `form` is actually the updated inputs which we want to override with (I'll update the naming actually for clarity). The return type of `NumberFieldBase` allows us to be certain that the shape has not changed

**Pull request merged**